PluginAutotask/API/Read/ReadRecords.cs

```diff
@@ -1,28 +1,120 @@
 using System;
 using System.Collections.Generic;
-using System.Threading.Tasks;
-using Naveego.Sdk.Plugins;
-using PluginHubspot.API.Factory;
-using PluginHubspot.API.Utility;
-using PluginHubspot.Helper;
+using Aunalytics.Sdk.Logging;
+using Aunalytics.Sdk.Plugins;
+using Newtonsoft.Json;
+using PluginAutotask.API.Factory;
+using PluginAutotask.DataContracts;
 
-namespace PluginHubspot.API.Read
+namespace PluginAutotask.API.Read
 {
     public static partial class Read
     {
-        public static async IAsyncEnumerable<Record> ReadRecordsAsync(IApiClient apiClient, Schema schema, Settings settings, DateTime? lastReadTime = null, TaskCompletionSource<DateTime>? tcs = null)
+        public static async IAsyncEnumerable<Record> ReadRecordsAsync(IApiClient apiClient, Schema schema, int limit = -1) 
         {
-            var endpoint = EndpointHelper.GetEndpointForSchema(schema);
-
-            var records = endpoint?.ReadRecordsAsync(apiClient, lastReadTime, tcs);
-
-            if (records != null)
+            if (schema.Id == "TicketHistory")
             {
+                var records = ReadRecordsTicketHistoryAsync(apiClient, schema, limit);
+                
                 await foreach (var record in records)
                 {
                     yield return record;
                 }
             }
+
+            var query = Utility.Utility.GetQueryForSchemaId(schema.Id);
+            if (limit >= 0) 
+            {
+                query.MaxRecords = Math.Min(limit, 500);
```

Comment:
Limit 500 could be in constants.cs as it has been used in two places, but not substantial.